### ISSUE TITLE
Do we need to escape search string as it's used in regexp? Wondering what's the result of `contains("abcdefg", ".*")`

### ISSUE THREAD

**Issue by alamb** [apache/datafusion#10929]
Do we need to escape search string as it's used in regexp? Wondering what's the result of `contains("abcdefg", ".*")`

_Originally posted by @waynexia in https://github.com/apache/datafusion/pull/10879#discussion_r1635767599_

**Comment by alamb** [comment]
cc @Lordworms

**Comment by Lordworms** [comment]
Sorry for the late review since I was busy this week. In the beginning, I was just trying to keep the same format as other ScalarUDF which utilize arrow-rs methods to implement functionality so I just chose arrow::reglike. I can fix it to use str.contains

**Comment by alamb** [comment]
> I was just trying to keep the same format as other ScalarUDF which utilize arrow-rs methods to implement functionality so I just chose arrow::reglike. 

Makes sense. I think in this case, however, the function shouldn't actually have regexp support, so it would be better to use str.contains

> I can fix it to use str.contains

Thank you!

**Comment by waynexia** [comment]
Thanks @Lordworms. I'm thinking about @alamb's idea https://github.com/apache/datafusion/pull/10879#discussion_r1636789004, which only implements `contains` as a placeholder for translating/planning. And it would finally become some other thing like `LIKE` after the optimization phase. A similar thing is `Expr::Wildcard`, it's to reflect `*` from SQL but doesn't have a corresponding physical expr.

**Comment by alamb** [comment]
One benefit of using `LIKE` is that it already has a highly optimized arrow implementation (as in will actually use substring if the patttern looks like `substr%` etc).

**Comment by Lordworms** [comment]
I'll refactor it to use LIKE
